src/Serialization/Rings.jl

```diff
@@ -226,19 +226,31 @@ end
 @registerSerializationType(nmod_mat)
 
 function save_internal(s::SerializerState, m::MatrixElem)
+    m_base_ring = save_type_dispatch(s, base_ring(parent(m)))
     return Dict(
         :matrix => save_type_dispatch(s, Array(m)),
+        :base_ring => m_base_ring
     )
 end
 
 function load_internal(s::DeserializerState,
                        ::Type{<: MatElem},
                        dict::Dict)
+    entries_ring = load_unknown_type(s, dict[:base_ring])
     mat = load_type_dispatch(s, Matrix, dict[:matrix])
-    entries_ring = parent(mat[1])
+
     return matrix(entries_ring, mat)
 end
 
+function load_internal_with_parent(s::DeserializerState,
+                                   ::Type{<: MatElem},
+                                   dict::Dict,
+                                   parent_ring::T) where T <: MatSpace
+    _ = load_unknown_type(s, dict[:base_ring])
```

Comment:
Why do we even load the ring here, when we don't intend to use it?

Reply:
this is similar to other methods where I load a parent or a base and don't use it. It happens when loading with a given parent, and it is due to certain backrefs being stored in the base and or parent. I'll add comment so this is clearer, and I make sure there are comment where it happens else where as well.